### PR TITLE
St33lphoenix/fixes

### PR DESCRIPTION
**Pull request by St33lPhoenix**
Add database configuration to config.yml
Add index file configuration to config.yml
Add local Maven repository location to config.yml
Support running multiple index files
Refactor indexer reader
Add logging to file
Add extractor for artifact presence
Add support for packages without archives

PSA:
Extractors can now see packages without artifact. Verify the extractor behaviour.
`SizeExtractorTest` is disabled due to buggy tests.

### CODE REVIEW

File: analyzer/src/main/java/nl/tudelft/Maven.java [hunk 1]
```diff
@@ -22,18 +30,30 @@ public Package getPackage(PackageId id, String pkgType) throws PackageException
 
         Artifact artifact = resolver.createArtifact(id.group(), id.artifact(), id.version());
 
+        File jar;
         try {
-            JarFile artifactFile = new JarFile(resolver.getJar(artifact, pkgType));
-            Model pomFile = resolver.loadPom(artifact);
+            jar = resolver.getJar(artifact, pkgType);
+        } catch (ArtifactResolutionException exception) {
+            LOGGER.warn("Could not fetch archive ({})", id, exception);
```
Comment (Vel1khan) [inline analyzer/src/main/java/nl/tudelft/Maven.java:37]: Can you add this log to the errortable table?

**Comment by Vel1khan** [comment]
LGTM, my extractor still works as expected

**Review by cryocz — APPROVED** [review]
cool

File: analyzer/src/main/java/nl/tudelft/Maven.java (hunk 1 above)
Comment (Vel1khan) [reply analyzer/src/main/java/nl/tudelft/Maven.java:37]: nvm, it's clear in the packages table whether there's no archive

**Pull request merged**